File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs[0]: return ""

        prefix = strs[0]
        
        for s in strs:
            newPrefix = ""
            idx = 0
            
            while (idx < len(s)) and (idx < len(prefix)) and (s[idx] == prefix[idx]):
                newPrefix += s[idx]
                idx += 1
            prefix = newPrefix
            
        return prefix

File: 14-longest-common-prefix/longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    if (!strs[0]) return "";

    let prefix: string = strs[0];

    for (const str of strs) {
        let newPrefix = "";
        let idx = 0;
        while (idx < str.length && idx < prefix.length && str[idx] === prefix[idx]) {
            newPrefix += str[idx];
            idx++;
        }
        
        prefix = newPrefix
    }

    return prefix;
};